src/main/java/hibernate/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -51,4 +51,16 @@ public CinemaHall add(CinemaHall cinemaHall) {
             throw new DataProcessingException("Can't get all cinema halls", e);
         }
     }
+
+    @Override
+    public CinemaHall get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<CinemaHall> query = session.createQuery("from CinemaHall"
+                    + " WHERE id = :id", CinemaHall.class);
+            query.setParameter("id", id);
+            return query.getSingleResult();
```

Comment:
```suggestion
           return session.get(CinemaHall.class, id);
```
you can use this method